src/core/format/IdGenerator.ts

```diff
@@ -23,30 +24,56 @@ export class IdGenerator {
      * @param {object} ownerPublicId The public id.
      * @returns {module:coders/uint64~uint64} The mosaic id.
      */
-    public static generateMosaicId = (nonce, ownerPublicId) => {
-        const hash = sha3_256.create();
+    public static generateMosaicId = (
+        nonce,
+        ownerPublicId,
+        signSchema: SignSchema = SignSchema.SHA3,
+    ) => {
+        const hash = SHA3Hasher.getHasher(32, signSchema).create();
         hash.update(nonce);
         hash.update(ownerPublicId);
         const result = new Uint32Array(hash.arrayBuffer());
         return [result[0], result[1] & 0x7FFFFFFF];
     }
 
+    /**
+     * Generates a namespace id given a parent id and name.
+     * @param {object} parentId The parent namespace id.
+     * @param {object} name The namespace name.
+     * @returns {module:coders/uint64~uint64} The namespace id.
+     */
+    public static generateNamespaceId = (
+        parentId,
+        name: string,
+        signSchema: SignSchema = SignSchema.SHA3,
+    ) => {
+        const hash = SHA3Hasher.getHasher(32, signSchema).create();
+        hash.update(Uint32Array.from(parentId).buffer as any);
+        hash.update(name);
+        const result = new Uint32Array(hash.arrayBuffer());
+        // right zero-filling required to keep unsigned number representation
+        return [result[0], (result[1] | 0x80000000) >>> 0];
+    }
+
     /**
      * Parses a unified namespace name into a path.
      * @param {string} name The unified namespace name.
      * @returns {array<module:coders/uint64~uint64>} The namespace path.
      */
-    public static generateNamespacePath = (name: string) => {
+    public static generateNamespacePath = (
+        name: string,
+        signSchema: SignSchema = SignSchema.SHA3,
```

Comment:
Having a default is error-prone. Delegate methods may miss that parameter and the user won't be able to set that param. It has happened before. 

Use NetworkType instead of SignSchema. It has better availability.

Reply:
you are reviewing old PR ..